pkg/sql/catalog/tabledesc/table.go

```diff
@@ -168,7 +168,7 @@ func MakeColumnDefDescs(
 		// Verify the on update expression type is compatible with the column type
 		// and does not contain invalid functions.
 		ret.OnUpdateExpr, err = schemaexpr.SanitizeVarFreeExpr(
-			ctx, d.OnUpdateExpr.Expr, resType, "ON UPDATE", semaCtx, volatility.Volatile,
+			ctx, d.OnUpdateExpr.Expr, resType, "ON UPDATE", semaCtx, volatility.Volatile, true,
```

Comment:
nit: It's helpful to label arguments with simple constants to make it more clear, like `true /* allowAssignmentCast */`.